classes/webservice/WebserviceOutputBuilder.php

```diff
@@ -711,7 +711,7 @@ protected function renderFlatAssociation($object, $depth, $assoc_name, $resource
 
         foreach ($fields_assoc as $field_name => $field) {
             if (!is_array($this->fieldsToDisplay) || in_array($field_name, $this->fieldsToDisplay[$assoc_name])) {
-                if ($field_name == 'id' && !isset($field['sqlId'])) {
+                if (isset($field['id']) && !isset($field['sqlId'])) {
```

Comment:
this would work prior to php8, because the condition `$field_name == 'id'` would evalute to **true** when `$field_name` was equal to zero:

https://www.php.net/releases/8.0/en.php#saner-string-to-number-comparisons

But it should never have worked, here we should evaluate that the field ID exists, if it doesn't exist then we should be in the `else if` case. 

So the bug was revealed with php8, but the code was always broken.